### ISSUE TITLE
Generate tree/list colors based on a single color

### ISSUE THREAD

**Issue by isidorn** [Microsoft/vscode#25417]
Refs #25328

It is quite complex to theme the explorer tree due to the following reasons
1. The name of the settings always starts with `list` and are hardly discoverable
2. There are 6 settings with very similar names and figuring out which one does what is a mayor pain
3. Some of the settings seem to be overlaping

This case is a clear example of how an automatic color picker would be helpful. I would just define the base color for my tree selection and then all the other would just be lighter / darker variantes of the base color.

The current way I have to understand what each of the settigns does and make sure I did not miss any case and there are a lot of those when customizing the tree.

**Comment by bpasero** [comment]
Yeah, tree is tough. I would also think that most colors could be derived, given a base color.

**Comment by sgtoj** [comment]
@isidorn Are you looking for something like [vscode-theme-generator](https://github.com/Tyriar/vscode-theme-generator)? It was used to generate [**Sapphire**](https://marketplace.visualstudio.com/items?itemName=Tyriar.theme-sapphire). The developer behind the tool a is [Daniel Imms](https://github.com/Tyriar), a VS Code developer at Microsoft.

**Comment by Tyriar** [comment]
@sgtoj the hope is eventually all the derived color rules move into VS Code as there's not much reason not to. The main problem the theme generator is aiming to tackle is the horrible .tmTheme editing experience, as well as being a testing group for the derived color rules :smiley:

**Comment by bpasero** [comment]
I will see if that would be possible.

**Comment by bpasero** [comment]
For April I was able to get rid of the following two colors, making the list/tree easier to adopt:
* `list.focusAndSelectionBackground`
* `list.focusAndSelectionForeground`

Generating the other colors is a bit tricky...

**Comment by Tyriar** [comment]
Related discussion: https://github.com/Microsoft/vscode/issues/25482#issuecomment-298773156

TLDR: Let's experiment in the generator first.
